### PR TITLE
[OTX] `pip install otx` phase 1: removing submodules and git+https dependencies

### PR DESCRIPTION
**Pull request by goodsong81**
[In this PR]
- Removed submodules: uploaded to PyPI and applied to requirements
  - MPA is the only remaining git+https requirement item (to be removed after code freeze)
  - OMZ Model APIs package was also uploaded to PyPI
- Current progress for `pip install otx[full]` is represented in otx/algorithms/init_venv.sh
  - Still needs pytorch as prerequisite; to be documented or handled separately
  - `pip install --use-deprecated=legacy-resolver -e ../../[full]`
  - Needs legacy dependency resolver until public mmcv update
  - OTX package itself is not yet built or uploaded

[How to set up]
```console
# Install pytorch -> otx[full]
$ bash ./otx/algorithms/init_venv.sh venv
$ source ./venv/bin/activate

# Optionally, you may want to modify MPA.
# Then, you can install MPA with soft links
$ git clone https://github.com/openvinotoolkit/model_preparation_algorithm.git && cd model_preparation_algorithms
$ git checkout -b new_branch origin/otx
$ pip install -e .
# Fix code -> test -> create PR to MPA otx branch
```

### CODE REVIEW

File: otx/algorithms/init_venv.sh [hunk 1]
```diff
@@ -86,47 +85,30 @@ else
...
-pip install mmpycocotools
-
-# Build NNCF extensions
-echo "Build NNCF extensions ..."
-python -c "import nncf"
+# * Prerequisite
+#   - numpy: mmpycocotool uses source distribution, setup.py imports numpy
+#   - torch: mmdet/seg are installed via source, setup.py imports torch
+# * Issue
+#  - Dependency resolusion is to slow
```
Comment (sungmanc) [inline otx/algorithms/init_venv.sh]: resolusion --> resolution

File: otx/algorithms/init_venv.sh [hunk 2]
```diff
@@ -86,47 +85,30 @@ else
...
+#  - Several lib version conflicts
+#  -> Temporary use of --use-deprecated=legacy-resolve
```
Comment (sungmanc) [inline otx/algorithms/init_venv.sh]: resolve --> resolver?

**Review by sungmanc — DISMISSED** [review]
LGTM, I just added some minor comments regarding typo

**Review by sungmanc — APPROVED** [review]

**Review by samet-akcay — APPROVED** [review]
Thanks, yet another important step done!

**Pull request merged**